idp/idputil/idputil.go

```diff
@@ -240,3 +240,19 @@ func ServiceURL(location, dest string) string {
 	lu.Path = path.Join(lu.Path, u.Path)
 	return lu.String()
 }
+
+// CookiePathRelativeToLocation returns the Login Cookie Path
+// relative to the sub-path in the location URL given.
+func CookiePathRelativeToLocation(cookiePath, location string, useLocation bool) string {
+	if useLocation == false {
+		return cookiePath
+	}
+
+	relativePath := ""
+	u, err := url.Parse(location)
+	if err == nil {
```

Comment:
or check `if err != nil` and `return u.Path + cookiePath`, so you can just drop the relativePath var?